core/src/main/scala/org/apache/spark/SparkContext.scala

```diff
@@ -1870,21 +1870,21 @@ class SparkContext(config: SparkConf) extends Logging {
       }
     }
 
-    if (path == null) {
-      logWarning("null specified as parameter to addJar")
+    if (path == null || path.isEmpty) {
+      logWarning("null or empty path specified as parameter to addJar")
     } else {
       val key = if (path.contains("\\")) {
         // For local paths with backslashes on Windows, URI throws an exception
         addLocalJarFile(new File(path))
       } else {
-        val uri = new URI(path)
+        val uri = new Path(path).toUri
```

Comment:
nit: `hadoopPath.toUri`

Reply:
hadooppath define inside `checkRemoteJarFile` so i am unable to used outside